### DISCUSSION TITLE
Support rfc6570 uritemplates in click handler

### DISCUSSION THREAD

**Discussion by dschulten** [Kong/insomnia#3735] [Ideas]
[Feature Request] – APIs can contain rfc6570 uritemplates, e.g. `http://example.com{?personId*}`. Building on https://github.com/getinsomnia/insomnia/issues/874, Insomnia should allow to provide uritemplate variables when the user clicks such a UriTemplate:

![image](https://user-images.githubusercontent.com/107194/68068181-f8832280-fd52-11e9-8c0e-4bd3d546d64e.png)

**Comment by dschulten** [comment]
Feature branch: https://github.com/dschulten/insomnia/tree/feature/follow-rfc6570-uritemplates

In uritemplates you can have [composite values](https://tools.ietf.org/html/rfc6570#section-2.4.2), e.g. lists or dictionaries. E.g. if `personId` is a list `[123,456]`, then the template `http://example.com{?personId*}` expands to `http://example.com?personId=123&personId=456`. The same works for objects like `{city: 'Newport Beach',state: 'CA'}`.
We can allow users to edit such composite values in the multiline editor. When multiline is enabled, a button **Click to edit** allows to open the editor instead of an input field.

Uritemplate variables are a much wider concept than url query params, although there is some overlap. We therefore keep them separate from `Request.params` and use them only temporarily to build complete urls.

**Comment by gschier** [comment]
I like this! Nice work. 

At first glance (just by the screenshot), it doesn't seem like this should depend on #874. Ideally it should be treated as completely separate and should even be able to be merged into the existing system of opening links in the browser.

Feel free to submit a PR for this, ideally without the included changes in #874
